libcaf_io/src/default_multiplexer.cpp

```diff
@@ -1217,6 +1218,26 @@ new_tcp_acceptor(uint16_t port, const char* addr, bool reuse) {
                                   bound_port};
 }
 
+std::string addr_of_fd(native_socket fd) {
+  sockaddr_storage st;
+  socklen_t st_len = sizeof(st);
+  sockaddr* sa = reinterpret_cast<sockaddr*>(&st);
+  ccall(cc_zero, "getsocketname() failed",
+        getsockname, fd, sa, &st_len);
+  char addr[40] {0};
```

Comment:
You could use `INET6_ADDRSTRLEN` instead of `40`, or explain in a comment why provisioning 40 will suffice in all scenarios. Otherwise this is a potential memory safety issue.